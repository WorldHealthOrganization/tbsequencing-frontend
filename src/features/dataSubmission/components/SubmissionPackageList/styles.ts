import { css } from '@emotion/react';

const PADDING = 54;

export const wrapper = css({
  paddingTop: 40,
  paddingLeft: PADDING,
  paddingRight: PADDING,
  paddingBottom: 24,
});

export const pageHeader = css({
  marginBottom: 24,
});

export const tableContainer = css({
  position: 'relative',
});
