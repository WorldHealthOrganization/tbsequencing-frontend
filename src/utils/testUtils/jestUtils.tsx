import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
