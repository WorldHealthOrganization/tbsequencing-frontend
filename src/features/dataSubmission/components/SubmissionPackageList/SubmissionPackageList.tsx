/** @jsxImportSource @emotion/react */
import React, { useState } from 'react';
import { SubmitHandler } from 'react-hook-form';
import { toast } from 'react-toastify';
import { useMsal } from '@azure/msal-react';
import SubmissionPackageListHeader from './components/SubmissionPackageListHeader';
import { useCreatePackageMutation, useGetPackagesQuery } from '../../../../services/submissionApi';
import SubmissionPackageListTable from './components/SubmissionPackageListTable';
import PackageModal from '../PackagePage/components/PackageModal';
import { IInputs } from '../PackagePage/components/PackageModal/PackageModal';
import BasePage from '../../../../components/BasePage';
import SubmissionPackageUserInfo from '../PackagePage/components/SubmissionPackageUserInfo';
import { tableContainer } from './styles';

export const SubmissionPackageList = () => {
  const [createModalOpened, setCreateModalOpened] = useState<boolean>(false);
  const openModal = () => setCreateModalOpened(true);
  const closeModal = () => setCreateModalOpened(false);
  const { data, isLoading } = useGetPackagesQuery();
  const [createPackage, { isLoading: isCreatePackageLoading }] = useCreatePackageMutation();
  const { accounts } = useMsal();

  const packagesEmpty = data?.length === 0;
  const shouldRenderPackageList = !packagesEmpty;

  const onCreatePackageClick = () => {
    openModal();
  };

  const handleSubmit: SubmitHandler<IInputs> = async (formData) => {
    try {
      await createPackage({ ...formData });
      closeModal();
    } catch (e) {
      toast.error('There was an error trying to create a package, try another name');
    }
  };

  return (
    <BasePage pageHeader="Submission Package List">
      { accounts[0] && (
        <SubmissionPackageUserInfo />
      )}

      <SubmissionPackageListHeader
        onCreateClick={onCreatePackageClick}
      />
      <PackageModal
        title="New Package"
        isLoading={isLoading || isCreatePackageLoading}
        okayLabel="Create Package"
        cancelLabel="Cancel"
        open={createModalOpened}
        onCancelClick={closeModal}
        onSubmit={handleSubmit}
      />
      {shouldRenderPackageList && (
      <div css={tableContainer}>
        <SubmissionPackageListTable loading={isLoading} packages={data} />
      </div>
      )}
    </BasePage>
  );
};

export default SubmissionPackageList;
