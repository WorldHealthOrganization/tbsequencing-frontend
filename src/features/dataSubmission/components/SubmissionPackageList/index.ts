import SubmissionPackageList from './SubmissionPackageList';

export default SubmissionPackageList;
