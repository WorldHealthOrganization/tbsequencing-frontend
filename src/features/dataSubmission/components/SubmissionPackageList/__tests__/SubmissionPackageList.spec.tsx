import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { getTreeJSONWithRedux, render, screen } from '../../../../../utils/testUtils/jestUtils';
import SubmissionPackageList from '../SubmissionPackageList';

describe('SubmissionPackageList component test', () => {
  const componentWrappedInRouter = (
    <BrowserRouter>
      <SubmissionPackageList />
    </BrowserRouter>
  );

  test('renders SubmissionPackageList component without error', () => {
    render(
      componentWrappedInRouter,
    );
  });

  it('Should render title properly', () => {
    render(
      componentWrappedInRouter,
    );

    screen.getByText('Submission Package List');
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      componentWrappedInRouter,
    );

    expect(tree).toMatchSnapshot();
  });
});
