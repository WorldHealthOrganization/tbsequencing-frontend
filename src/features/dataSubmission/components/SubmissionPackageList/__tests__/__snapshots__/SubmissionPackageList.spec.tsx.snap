// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionPackageList component test Should render correctly 1`] = `
<div
  className="css-xn17ji-BasePage"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-177n4ga-MuiTypography-root-BaseText"
  >
    Submission Package List
  </p>
  <div
    className="css-soo0gy-SubmissionPackageListHeader"
  >
    <div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-15zbhfz-MuiButtonBase-root-MuiButton-root-AppButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
        >
          Create submission package
        </p>
      </button>
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root css-13a7vru-MuiButtonBase-root-MuiButton-root-AppButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
        >
          Terms & Conditions
        </p>
      </button>
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1coqdjr-MuiButtonBase-root-MuiButton-root-AppButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
        >
          Instructions
        </p>
      </button>
    </div>
  </div>
  <div
    className="css-1pi7uhp-SubmissionPackageList"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1ag5kwf-MuiCircularProgress-root"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </span>
  </div>
</div>
`;
